nilearn/image/image.py

```diff
@@ -502,6 +502,51 @@ def swap_img_hemispheres(img):
 
     # create swapped nifti object
     out_img = nibabel.Nifti1Image(img.get_data()[::-1], img.get_affine(),
-        header=img.get_header())
+                                  header=img.get_header())
 
     return out_img
+
+
+def index_img(imgs, index, check_imgs=True):
+    """Indexes into a 4D Niimg-like object in the fourth dimension
+
+    Parameter
+    ---------
+    imgs: 4D Niimg-like object
+          See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+
+    index: Any type compatible with numpy array indexing
+           Used for indexing the 4D data array in the fourth dimension
+
+    check_imgs: bool
+                Whether to check that `imgs` is a 4D Niimg-like object
+
+    Returns
+    -------
+    output: nibabel.Nifti1IMage
+            4D image whose data is a subset of the original image data
+
+    """
+    if check_imgs:
+        imgs = check_niimgs(imgs)
```

Comment:
I think we should not make this option available to people. You can create an `_index_img` method that does not do the `check_niimg` and call it from both functions.